storage/storage.go

```diff
@@ -197,6 +198,24 @@ func NewTupleKeyObjectIterator(tupleKeys []*openfgapb.TupleKey) ObjectIterator {
 	return NewStaticObjectIterator(objects)
 }
 
+func NewObjectIteratorFromTupleKeyIterator(iter TupleKeyIterator) ObjectIterator {
```

Comment:
Can we please implement this in a way that doesn't require the extra memory to statically store the list of objects?

Instead, we should implement the `ObjectIterator` interface with a struct that uses the `TupleKeyIterator` and yields the objects from that iterator dynamically. For example,

```
type tupleKeyObjectIterator struct {
    iter TupleKeyIterator
}

var _ ObjectIterator = &tupleKeyObjectIterator{} // make sure we implement the ObjectIterator interface

func (t *tupleKeyObjectIterator) Next() (*openfgapb.Object, error) {
    tk, err := t.iter.Next()
    if err != nil {
        return err
    }

    return tk.GetObject(), nil
}
```
then this function simply becomes
```
func NewObjectIteratorFromTupleKeyIterator(iter TupleKeyIterator) ObjectIterator {
    return &tupleKeyObjectIterator{iter}
}
```

Reply:
Good idea. Thanks!